packages/@webex/plugin-meetings/test/unit/spec/media/MediaConnectionAwaiter.ts

```diff
@@ -0,0 +1,326 @@
+      listener();
+      await testUtils.flushPromises();
+
+      assert.equal(promiseResolved, true);
+      assert.equal(promiseRejected, false);
+
+      // check that listener was removed
+      assert.calledTwice(mockMC.off);
+
+      assert.calledOnce(clearTimeoutSpy);
+    });
+
+    it(`ice gathering state update to 'gathering' state does not update timer`, async () => {
+      mockMC.getConnectionState.returns(ConnectionState.Connecting);
+      mockMC.getIceGatheringState.returns('new');
+
+      const clearTimeoutSpy = sinon.spy(clock, 'clearTimeout');
+
+      let promiseResolved = false;
+      let promiseRejected = false;
+
+      mediaConnectionAwaiter
+        .waitForMediaConnectionConnected()
+        .then(() => {
+          promiseResolved = true;
+        })
+        .catch(() => {
+          promiseRejected = true;
+        });
+
+      assert.equal(promiseResolved, false);
+      assert.equal(promiseRejected, false);
+
+      // check the right listener was registered
+      assert.calledTwice(mockMC.on);
+      assert.equal(mockMC.on.getCall(0).args[0], Event.CONNECTION_STATE_CHANGED);
+      assert.equal(mockMC.on.getCall(1).args[0], Event.ICE_GATHERING_STATE_CHANGED);
+      const listener = mockMC.on.getCall(1).args[1];
+
+      // call the listener and pretend we are now connected
+      mockMC.getIceGatheringState.returns('gathering');
+      listener();
+
+      mockMC.getConnectionState.returns(ConnectionState.Connected);
+
+      await clock.tickAsync(ICE_AND_DTLS_CONNECTION_TIMEOUT);
+      await testUtils.flushPromises();
+
+      assert.equal(promiseResolved, true);
+      assert.equal(promiseRejected, false);
+
+      // check that listener was removed
+      assert.calledTwice(mockMC.off);
+
+      assert.neverCalledWith(clearTimeoutSpy);
+    });
+
+    it(`ice gathering update to 'complete' state update timer`, async () => {
+      mockMC.getConnectionState.returns(ConnectionState.Connecting);
+      mockMC.getIceGatheringState.returns('new');
+
+      const clearTimeoutSpy = sinon.spy(clock, 'clearTimeout');
+
+      let promiseResolved = false;
+      let promiseRejected = false;
+
+      mediaConnectionAwaiter
+        .waitForMediaConnectionConnected()
+        .then(() => {
+          promiseResolved = true;
+        })
+        .catch(() => {
+          promiseRejected = true;
+        });
+
+      assert.equal(promiseResolved, false);
+      assert.equal(promiseRejected, false);
+
+      // check the right listener was registered
+      assert.calledTwice(mockMC.on);
+      assert.equal(mockMC.on.getCall(0).args[0], Event.CONNECTION_STATE_CHANGED);
+      assert.equal(mockMC.on.getCall(1).args[0], Event.ICE_GATHERING_STATE_CHANGED);
+      const listener = mockMC.on.getCall(1).args[1];
+
+      // call the listener and pretend we are now connected
+      mockMC.getIceGatheringState.returns('complete');
+      listener();
+
+      mockMC.getConnectionState.returns(ConnectionState.Connected);
```

Comment:
do we have a test where we get gathering complete callback followed by a connection connected callback, all within the timeout? it's almost the same as this one here, but this one doesn't fire the connection state change callback and that callback will actually be called in most cases in real life, right?

Reply:
Added test for this case